github/repos_rules.go

```diff
@@ -943,25 +956,21 @@ func (s *RepositoriesService) UpdateRuleset(ctx context.Context, owner, repo str
 // GitHub API docs: https://docs.github.com/rest/repos/rules#update-a-repository-ruleset
 //
 //meta:operation PUT /repos/{owner}/{repo}/rulesets/{ruleset_id}
-func (s *RepositoriesService) UpdateRulesetNoBypassActor(ctx context.Context, owner, repo string, rulesetID int64, rs *Ruleset) (*Ruleset, *Response, error) {
+func (s *RepositoriesService) UpdateRulesetNoBypassActor(ctx context.Context, owner, repo string, rulesetID int64, rs Ruleset) (*Ruleset, *Response, error) {
```

Comment:
@gmlewis my understanding of the code here is that for all ruleset updates we should just use the `rulesetNoOmitBypassActors` type if the bypass actors is empty. Does that make sense and is there a documented pattern for this already established?

Reply:
Yes, that makes sense and if you would prefer to use this pattern, that is also fine by me if you feel that makes sense for your endpoints (meaning passing a `nil` has a a documented, predefined meaning).

Note that you should document what happens in your endpoints if `nil` is passed.

And since this endpoint already documents what happens when `nil` is passed, there is really no need to make a breaking change to this endpoint.

Reply:
@gmlewis this has been done. Could you please re-review?